nats.go

```diff
@@ -2507,8 +2507,10 @@ func (nc *Conn) IsConnected() bool {
 	return nc.isConnected()
 }
 
-// Servers returns the list of known servers, including additional
-// servers discovered after a connection has been established.
+// Servers returns the list of known server urls, including additional
+// servers discovered after a connection has been established.  URLs passed
+// into NATS through the API be preserved in their original form, but any
+// credentials in a URL will never be visible outside of the application.
```

Comment:
I think that if the user specifies the URLs with credentials through Options.Servers (or the connect URL), I don't see a problem returning those URLs with credentials (as they were entered). The one discovered will not have credentials. So I don't see the need to distinguish between explicit vs implicit.